From b39ed0b3bf12bc8e90a15cda43f7ccade7d79be1 Mon Sep 17 00:00:00 2001
From: Amruta Kulkarni <amrukulk@qti.qualcomm.com>
Date: Wed, 30 Jul 2025 13:44:37 -0700
Subject: [PATCH 18/40] qcacld-3.0: Add check to avoid buffer overflow

Currently in api cm_roam_stats_print_scan_info(),
out of bound write can possibly happen.

To avoid this add check to avoid buffer overflow.

CRs-Fixed: 4217059
Change-Id: I8e67138c0e882a852cd33b275bc3202876d66a10
---
 .../mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c   | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c b/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c
index f946829699..4470db4890 100644
--- a/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c
+++ b/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c
@@ -3701,10 +3701,14 @@ cm_roam_stats_print_scan_info(struct wlan_objmgr_psoc *psoc,
 		buf_cons = qdf_snprint(tmp, buf_left, "{");
 		buf_left -= buf_cons;
 		tmp += buf_cons;
-
 		for (i = 0; i < num_ch; i++) {
 			buf_cons = qdf_snprint(tmp, buf_left, "%d ",
 					       scan->chan_freq[i]);
+			if (buf_cons > buf_left) {
+				mlme_err("buf_left size insufficient");
+				qdf_mem_free(buf);
+				return;
+			}
 			buf_left -= buf_cons;
 			tmp += buf_cons;
 		}
-- 
2.34.1

